### ISSUE TITLE
Multi-mat variable on accelerator: same view to access global value and partial value 

### ISSUE THREAD

**Issue by DavidDureau** [arcaneframework/framework#859]
With the same view, we would like to access a global value (through a `CellLocalId`) or a partial value (through a `MatVarIndex`):
```c++
auto out_menv_var1   = ax::viewOut  (command, m_menv_var1);
auto in_menv_var2    = ax::viewIn   (command, m_menv_var2);
auto inout_menv_var3 = ax::viewInOut(command, m_menv_var3);

CellToAllEnvCellAccessor cell2allenvcell(m_mesh_material_mng);

command << RUNCOMMAND_ENUMERATE_CELL_ALLENVCELL(cell2allenvcell, cid, allCells()) {

  Real sum2=0.;
  ENUMERATE_CELL_ALLENVCELL(iev, cid, cell2allenvcell) {
    sum2 += in_menv_var2[*iev]/in_menv_var2[cid];
  }    

  Real sum3=0.;
  ENUMERATE_CELL_ALLENVCELL(iev, cid, cell2allenvcell) {
    Real contrib2 = (in_menv_var2[*iev]/in_menv_var2[cid])*(sum2+1.);
    inout_menv_var3[*iev] = contrib2 * inout_menv_var3[cid];
    sum3 += contrib2;
  }    

  out_menv_var1[cid] = sum3;
}; 
```

**Comment by DavidDureau** [comment]
I can't compile this kind of code:
```c++
MaterialVariableCellReal var;

auto out_var = ax::viewOut(command, var);
command << RUNCOMMAND_ENUM..(Cell, cid, allCells())
{
  out_var[cid] = value;
};
```

The bug seems to be at that line:

https://github.com/arcaneframework/framework/blob/d6e88b50d9e5aca7e50ba8b4332b006c92d3c92e/arcane/src/arcane/accelerator/MaterialVariableViews.h#L151

The type `const DataType&`  should be replaced by the type `Accessor`.

**Comment by francois-letierce** [comment]
fixed with #981
